From: Eric Blake <eblake@redhat.com>
Date: Tue, 24 Jan 2012 18:51:01 +0000 (-0700)
Subject: lxc: export container=lxc-libvirt for systemd
X-Git-Url: http://libvirt.org/git/?p=libvirt.git;a=commitdiff_plain;h=16dc4ade7a2c83f166c7e8b8560a5fd42e662b53

lxc: export container=lxc-libvirt for systemd

Systemd detects containers based on whether they have
an environment variable starting with 'container=lxc';
using a longer name fits the expectations, while also
allowing detection of who created the container.

Requested by Lennart Poettering, in response to
https://bugs.freedesktop.org/show_bug.cgi?id=45175

* src/lxc/lxc_container.c (lxcContainerBuildInitCmd): Add another
env-var.
---

diff --git a/src/lxc/lxc_container.c b/src/lxc/lxc_container.c
index dcd65ef..6eb188d 100644
--- a/src/lxc/lxc_container.c
+++ b/src/lxc/lxc_container.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2008-2011 Red Hat, Inc.
+ * Copyright (C) 2008-2012 Red Hat, Inc.
  * Copyright (C) 2008 IBM Corp.
  *
  * lxc_container.c: file description
@@ -119,6 +119,7 @@ static virCommandPtr lxcContainerBuildInitCmd(virDomainDefPtr vmDef)
 
     virCommandAddEnvString(cmd, "PATH=/bin:/sbin");
     virCommandAddEnvString(cmd, "TERM=linux");
+    virCommandAddEnvString(cmd, "container=lxc-libvirt");
     virCommandAddEnvPair(cmd, "LIBVIRT_LXC_UUID", uuidstr);
     virCommandAddEnvPair(cmd, "LIBVIRT_LXC_NAME", vmDef->name);
     if (vmDef->os.cmdline)
